modules/nodes-secondary-scheduler-configuring-console.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * nodes/scheduling/secondary_scheduler/nodes-secondary-scheduler-configuring.adoc
+
+:_content-type: PROCEDURE
+[id="nodes-secondary-scheduler-configuring-console_{context}"]
+= Deploying a secondary scheduler
+
+After you have installed the {secondary-scheduler-operator}, you can deploy a secondary scheduler.
+
+.Prerequisities
+
+* You have access to the cluster with `cluster-admin` privileges.
+* You have access to the {product-title} web console.
+* The {secondary-scheduler-operator-full} is installed.
+
+.Procedure
+
+. Log in to the {product-title} web console.
+. Create config map to hold the configuration for the secondary scheduler.
+.. Navigate to *Workloads* -> *ConfigMaps*.
+.. Click *Create ConfigMap*.
+.. In the YAML editor, enter the config map definition that contains the necessary `KubeSchedulerConfiguration` configuration:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "secondary-scheduler-config"                  <1>
+  namespace: "openshift-secondary-scheduler-operator" <2>
+data:
+  "config.yaml": |
+    apiVersion: kubescheduler.config.k8s.io/v1beta1
+    kind: KubeSchedulerConfiguration                  <3>
+    leaderElection:
+      leaderElect: false
+    profiles:
+      - schedulerName: secondary-scheduler            <4>
+        plugins:                                      <5>
+          score:
+            disabled:
+              - name: NodeResourcesBalancedAllocation
+              - name: NodeResourcesLeastAllocated
+----
+<1> The name of the config map. This is used in the *Scheduler Config* field when creating the `SecondaryScheduler` CR.
+<2> The config map must be created in the `openshift-secondary-scheduler-operator` namespace.
+<3> The `KubeSchedulerConfiguration` resource for the secondary scheduler. For more information, see link:https://kubernetes.io/docs/reference/config-api/kube-scheduler-config.v1beta2/#kubescheduler-config-k8s-io-v1beta2-KubeSchedulerConfiguration[`KubeSchedulerConfiguration`] in the Kubernetes API documentation.
+// TODO: This link is v1beta2 though and our example is using v1beta1...
```

Comment:
@ingvagabund Question, I was trying to link for more info on KubeSchedulerConfiguration, but I don't see in in our OCP API docs, and the only place I could find in Kube was for v1beta2, and we look to be using v1beta1.

So first... is v1beta1 correct for us? And if so, then perhaps I can't use this link, unless there is a v1beta1 version somewhere that I can't see.

Reply:
v1beta1 is gone, upstream is on v1beta3. We need to be at least v1beta2. v1beta3 is the best.

Reply:
@ingvagabund Okay so I updated the above example to use `v1beta3` instead and am using the link to the v1beta3 docs too. Let me know if that sounds good.

FYI @kasturinarra because the test cases use v1beta1, in case those need to be updated/retested with v1beta3.

Reply:
@bergerhoffer thanks, i will update the test cases to have beta3 and test them once as well, thanks !!

Reply:
Looks good

Reply:
@ingvagabund @bergerhoffer i see having v1beta3 does not work and gives me below error in the cluster pod logs in openshift-secondary-scheduler-operator namespace.
I0412 09:41:57.501878       1 serving.go:347] Generated self-signed cert in-memory
no kind "KubeSchedulerConfiguration" is registered for version "kubescheduler.config.k8s.io/v1beta3" in scheme "k8s.io/kubernetes/pkg/scheduler/apis/config/scheme/scheme.go:31"
[knarra@knarra ~]$ oc get pods -n openshift-secondary-scheduler-operator
NAME                                            READY   STATUS             RESTARTS      AGE
secondary-scheduler-5d9d9d7c99-9vlfx            0/1     CrashLoopBackOff   2 (23s ago)   61s
secondary-scheduler-operator-6dd5ddc4c8-cbtkz   1/1     Running            0             11m

Reply:
@ingvagabund k8s.gcr.io/scheduler-plugins/kube-scheduler:v0.22.6

Reply:
1.22 does not have v1beta2: https://github.com/kubernetes/kubernetes/tree/release-1.22/pkg/scheduler/apis/config. 1.23 does have it: https://github.com/kubernetes/kubernetes/tree/release-1.23/pkg/scheduler/apis/config. https://github.com/kubernetes-sigs/scheduler-plugins#compatibility-matrix still has not released 1.23 version yet. v1beta2 will need to do it for now. Though, the documentation can stay with v1beta3 since it's user configurable.